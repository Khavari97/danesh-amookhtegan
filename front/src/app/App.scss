@import 'assets/fonts/fonts';

* {
  margin: 0;
  font-family: "Shabnam", serif;
}

a {
  text-decoration: none;
}

//Slider css
.alice-carousel__dots {
  .alice-carousel__dots-item:not(.__custom).__active {
    background-color: #046B00;
  }
}


