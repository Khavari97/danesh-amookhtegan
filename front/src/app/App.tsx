import React from "react";
import "./App.scss";
import AppRouter from "./navigation/AppRouter";
import {SnackbarProvider} from 'notistack';
import {SnackbarUtils} from "./components/material-ui/Snackbar/SnackbarUtils";
import ThemeProvider from "./assets/styles/material/material";

export interface AppProps {
}

export interface AppState {
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {};
    }

    componentDidMount = () => {
    }

    render() {
        return (
            <SnackbarProvider maxSnack={3} anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}>
                <SnackbarUtils/>
                <ThemeProvider>
                    <AppRouter/>
                </ThemeProvider>
            </SnackbarProvider>
        );
    }
}

export default App;
