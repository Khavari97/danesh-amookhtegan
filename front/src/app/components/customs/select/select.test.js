import React from "react"
import App from "./select";
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { configure } from "enzyme";
import { render, waitFor,screen } from '@testing-library/react'
import UserEvent from "@testing-library/user-event";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'
import renderer from "react-test-renderer";
import { Provider } from "react-redux";

const middlewares = [thunk]
const mockStore = configureStore(middlewares);


configure({ adapter: new Adapter() })


configure({ adapter: new Adapter() })

const mockCallback = jest.fn();

describe(" test of ProjectResult page ", () => {
    let object = {
        label: 'string',
        valueId: 'any',
        data_test:'select-input',
        items: [{
            code: 3, id: 1,
            name: "tehran",
        },
        {
            code: 4, id: 2,
            name: "shiraz",
        }
        ],
        onChange: mockCallback
    }
    let store_mock = mockStore({})

     it('check call onchange select', async () => {
        const { getByRole, getByTestId, getByText, container } = render(
            <Provider store={store_mock}>
                <App object={object} />
            </Provider>
        );
        const selectNode = getByTestId("select-input");
        const selectButton = getByRole("button");
        expect(selectButton).not.toBeNull();
        expect(selectNode).not.toBeNull();
        UserEvent.click(selectButton);
        await waitFor(() => getByText("tehran"), { container });
        const itemClickable = getByText("tehran");
        UserEvent.click(itemClickable);
        expect(mockCallback).toHaveBeenCalled();
    });
    it('check call onchange select', async () => {
        const { getByRole, getByTestId, getByText, container } = render(
            <Provider store={store_mock}>
                <App object={object} />
            </Provider>
        );
        const selectNode = getByTestId("select-input");
        const selectButton = getByRole("button");
        expect(selectButton).not.toBeNull();
        expect(selectNode).not.toBeNull();
        UserEvent.click(selectButton);
        await waitFor(() => getByText("shiraz"), { container });
        const itemClickable = getByText("shiraz");
        UserEvent.click(itemClickable);
        expect(mockCallback).toHaveBeenCalled();
    });

    it('snapshot',()=>{
        const shot = (<Provider store={store_mock}>
        <App object={object} />
    </Provider>)
        const snapshot = renderer.create(shot).toJSON();
        expect(snapshot).toMatchSnapshot();
    })
})


