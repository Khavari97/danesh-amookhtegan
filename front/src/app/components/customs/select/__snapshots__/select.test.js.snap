// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` test of ProjectResult page  snapshot 1`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-13"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
    data-shrink={false}
    id="demo-simple-select-outlined-label"
  >
    string
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    data-testid="select"
    onClick={[Function]}
    onSelect={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="demo-simple-select-outlined-label mui-component-select-age"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
      id="mui-component-select-age"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      data-testid="select-input"
      id="demo-controlled-open-select"
      name="age"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-15 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-17"
      >
        <span>
          string
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
